dali/operators/bbox/bbox_rotate.cc

```diff
@@ -0,0 +1,388 @@
+ * @param inout_boxes ltrb img-coord boxes which are modified inplace
+ * @param remove_threshold threshold of area change to remove boxes
+ * @param image_wh width and height of the image in pixels
+ * @param in_labels optional input labels to filter synchronously with truncated boxes
+ * @param out_labels option result of the filtered labels
+ * @return the new number of valid boxes remaining
+ */
+int ClipAndRemoveBoxes(dali::span<vec4> inout_boxes, float remove_threshold, vec2 image_wh,
+                       std::optional<ConstSampleView<CPUBackend>> in_labels,
+                       std::optional<SampleView<CPUBackend>> out_labels) {
+  DALI_ASSERT(in_labels.has_value() == out_labels.has_value());
+  const bool handle_labels = in_labels.has_value();
+  if (handle_labels) {
+    DALI_ASSERT(in_labels->shape()[0] == inout_boxes.size());
+    DALI_ASSERT(out_labels->shape()[0] == inout_boxes.size());
+  }
+
+  int out_idx = 0;
+  for (int in_idx = 0; in_idx < inout_boxes.size(); ++in_idx) {
+    auto box = inout_boxes[in_idx];
+    const float old_area = (box.z - box.x) * (box.w - box.y);
+    box.x = std::clamp(box.x, 0.0f, image_wh.x);
+    box.y = std::clamp(box.y, 0.0f, image_wh.y);
+    box.z = std::clamp(box.z, 0.0f, image_wh.x);
+    box.w = std::clamp(box.w, 0.0f, image_wh.y);
+    const float new_area = (box.z - box.x) * (box.w - box.y);
+    const float fraction_remain = new_area / old_area;
+    if (fraction_remain >= remove_threshold) {
+      inout_boxes[out_idx] = box;
+      if (handle_labels) {
+        out_labels.value().mutable_data<int>()[out_idx] = in_labels.value().data<int>()[in_idx];
+      }
+      ++out_idx;
+    }
+  }
+  return out_idx;
+}
+
+int RotateBoxesKernel(ConstSampleView<CPUBackend> in_box_tensor,
+                      SampleView<CPUBackend> rotate_buffer, SampleView<CPUBackend> out_box_tensor,
+                      float angle, std::optional<ConstSampleView<CPUBackend>> in_labels,
+                      std::optional<SampleView<CPUBackend>> out_labels, bool ltrb, bool keep_size,
+                      Mode mode, float remove_threshold, vec2 image_wh, bool bbox_norm) {
+  const auto num_boxes = in_box_tensor.shape()[0];
+  auto in_boxes = dali::span(reinterpret_cast<const vec4*>(in_box_tensor.raw_data()), num_boxes);
+  auto x_coords = dali::span(rotate_buffer.mutable_data<float>(), 4 * num_boxes);
+  auto y_coords = dali::span(x_coords.end(), 4 * num_boxes);
+  if (ltrb) {
+    ExpandToAllCorners<true>(x_coords, y_coords, in_boxes);
+  } else {
+    ExpandToAllCorners<false>(x_coords, y_coords, in_boxes);
+  }
+
+  // Center the coordinates on zero in image coordinates
+  if (bbox_norm) {
+    std::transform(x_coords.begin(), x_coords.end(), x_coords.begin(),
+                   [w = image_wh.x](float elem) { return (elem - 0.5) * w; });
+    std::transform(y_coords.begin(), y_coords.end(), y_coords.begin(),
+                   [h = image_wh.y](float elem) { return (elem - 0.5) * h; });
```

Comment:
```suggestion
                   [w = image_wh.x](float elem) { return (elem - 0.5f) * w; });
    std::transform(y_coords.begin(), y_coords.end(), y_coords.begin(),
                   [h = image_wh.y](float elem) { return (elem - 0.5f) * h; });
```
0.5 results in an unnecessary promotion to double and poor assembly.